src/config_builder.py

```diff
@@ -131,14 +136,23 @@ def _server(self) -> dict:
 
     @property
     def _storage_config(self) -> dict:
+        # Ref: https://grafana.com/docs/loki/latest/configure/#storage_config
         return {
-            "boltdb": {"directory": BOLTDB_DIR},
+            "boltdb_shipper": {
+                "active_index_directory": BOLTDB_DIR,
+                "shared_store": "filesystem",
+                "cache_location": BOLTDB_CACHE_DIR,
+            },
             "filesystem": {"directory": CHUNKS_DIR},
         }
 
     @property
     def _limits_config(self) -> dict:
         # Ref: https://grafana.com/docs/loki/latest/configure/#limits_config
+        if self.retention_period != 0:
```

Comment:
Add comment explaining the conversion?

Reply:
conversion actually seems redundant after another test, setting it to `0d` will yield same result so I removed the check